common/lib/xmodule/xmodule/js/src/javascript_loader.coffee

```diff
@@ -66,5 +66,52 @@ class @JavascriptLoader
         # just call the completion callback directly, without reloading the file
         completionHandlerGenerator(index)()
       $(placeholder).remove()
-        
-      
+
+  # loadXBlockResources:
+  #   With the provided XBlock resources:
+  #     1) Explicitly attach the resource to the <head> of document
+  #     2) Explicitly wait for each resource to be loaded
+  #     3) Return to callback function when all resources loaded
+  @loadXBlockResources: (resources, callback=null) ->
```

Comment:
Thanks @arbrandes. There was no way for you to know of this duplicated logic. You would only have known if you'd worked on Studio too. I like the idea of moving as much of this code as possible into common/xblock precisely so that it is centralized and folks don't have to dig through the platform to find it.

BTW, I assume that the conditional xblock just works in Studio. Is that correct?